lib/sqlalchemy/sql/elements.py

```diff
@@ -4340,11 +4458,11 @@ def _tq_label(self) -> Optional[str]:
         return self._gen_tq_label(self.name)
 
     @HasMemoized.memoized_attribute
-    def _render_label_in_columns_clause(self):
+    def _render_label_in_columns_clause(self) -> bool:
         return True
 
     @HasMemoized.memoized_attribute
-    def _non_anon_label(self):
+    def _non_anon_label(self) -> Optional[str]:
```

Comment:
Replacing `@HasMemoized.memoized_attribute` with `@property` resolves the error.